pkg/proxy/metrics/metrics.go

```diff
@@ -126,6 +126,17 @@ var (
 		},
 	)
 
+	// IptablesPartialRestoreFailuresTotal is the number of iptables *partial* restore
+	// failures (resulting in a fall back to a full restore) that the proxy has seen.
+	IptablesPartialRestoreFailuresTotal = metrics.NewCounter(
+		&metrics.CounterOpts{
+			Subsystem:      kubeProxySubsystem,
+			Name:           "sync_proxy_rules_iptables_partial_restore_failures_total",
+			Help:           "Cumulative proxy iptables partial restore failures",
+			StabilityLevel: metrics.ALPHA,
+		},
+	)
+
```

Comment:
maybe it is easier if we label existing metric
```
diff --git a/pkg/proxy/metrics/metrics.go b/pkg/proxy/metrics/metrics.go
index f61cf4fc991..6fb2089bdc7 100644
--- a/pkg/proxy/metrics/metrics.go
+++ b/pkg/proxy/metrics/metrics.go
@@ -117,13 +117,14 @@ var (
 
        // IptablesRestoreFailuresTotal is the number of iptables restore failures that the proxy has
        // seen.
-       IptablesRestoreFailuresTotal = metrics.NewCounter(
+       IptablesRestoreFailuresTotal = metrics.NewCounterVec(
                &metrics.CounterOpts{
                        Subsystem:      kubeProxySubsystem,
                        Name:           "sync_proxy_rules_iptables_restore_failures_total",
                        Help:           "Cumulative proxy iptables restore failures",
                        StabilityLevel: metrics.ALPHA,
                },
+               []string{"partial_sync", "full_sync"},
        )
```

Reply:
isn't that an API break of the existing metric? (I mean, ok, `StabilityLevel: metrics.ALPHA`, but still)

Reply:
I honestly don't know, I thought it was compatible, but better ask an expert
@dgrisonnet can you please advise? Is adding labels to an existing metric backwards compatible?

Reply:
@logicalhan can you help us with this question? is adding a label to a metric without labels compatible?

Reply:
It is a breaking change and you should probably amend the release note accordingly. But it is allowed since the metric is only `Alpha`.